contracts/stark-proxy/v2/impl/SP2Guardian.sol

```diff
@@ -0,0 +1,268 @@
+   */
+  function guardianRepayDebt(
+    uint256 amount
+  )
+    external
+    nonReentrant
+    onlyRole(GUARDIAN_ROLE)
+  {
+    _repayDebt(amount, true);
+  }
+
+  /**
+   * @notice Guardian method to trigger a withdrawal. This will transfer funds from StarkPerpetual
+   *  to this contract. This requires a (slow) withdrawal from L2 to have been previously processed.
+   *
+   *  Note: This function is intentionally not protected by the onlyAllowedKey modifier.
+   *
+   * @return The ERC20 token amount received by this contract.
+   */
+  function guardianWithdrawFromExchange(
+    uint256 starkKey,
+    uint256 assetType
+  )
+    external
+    nonReentrant
+    onlyRole(GUARDIAN_ROLE)
+    returns (uint256)
+  {
+    return _withdrawFromExchange(starkKey, assetType, true);
+  }
+
+  /**
+   * @notice Guardian method to trigger a forced withdrawal request.
+   *  Reverts if the borrower has no overdue debt.
+   *
+   *  Note: This function is intentionally not protected by the onlyAllowedKey modifier.
+   */
+  function guardianForcedWithdrawalRequest(
+    uint256 starkKey,
+    uint256 vaultId,
+    uint256 quantizedAmount,
+    bool premiumCost
+  )
+    external
+    nonReentrant
+    onlyRole(GUARDIAN_ROLE)
+  {
+    require(
+      getDebtBalance() > 0,
+      'SP2Guardian: Cannot call forced action if borrower has no overdue debt'
+    );
+    _forcedWithdrawalRequest(
+      starkKey,
+      vaultId,
+      quantizedAmount,
+      premiumCost,
+      true // isGuardianAction
+    );
+  }
+
+  /**
+   * @notice Guardian method to trigger a forced trade request.
+   *  Reverts if the borrower has no overdue debt.
+   *
+   *  Note: This function is intentionally not protected by the onlyAllowedKey modifier.
+   */
+  function guardianForcedTradeRequest(
+    uint256[12] calldata args,
+    bytes calldata signature
+  )
+    external
+    nonReentrant
+    onlyRole(GUARDIAN_ROLE)
+  {
+    require(
+      getDebtBalance() > 0,
+      'SP2Guardian: Cannot call forced action if borrower has no overdue debt'
+    );
+    _forcedTradeRequest(args, signature, true);
+  }
+
+  /**
+   * @notice Guardian method to prevent queued forced trade requests from being executed.
+   *
+   *  May only be called by VETO_GUARDIAN_ROLE.
+   *
+   * @param  argsHashes  An array of hashes for each forced trade request to veto.
+   */
+  function guardianVetoForcedTradeRequests(
+    bytes32[] calldata argsHashes
+  )
+    external
+    nonReentrant
+    onlyRole(VETO_GUARDIAN_ROLE)
+  {
+    for (uint256 i = 0; i < argsHashes.length; i++) {
+      bytes32 argsHash = argsHashes[i];
+      _QUEUED_FORCED_TRADE_TIMESTAMPS_[argsHash] = 0;
+      emit GuardianVetoedForcedTradeRequest(argsHash);
+    }
+  }
+
+  /**
+   * @notice Guardian method to request to cancel a pending deposit to the exchange.
+   *
+   * @param  starkKey   The STARK key of the account.
+   * @param  assetType  The exchange asset ID for the asset to deposit.
```

Comment:
I thought `The exchange asset ID for the deposit.` would be more clear than `The exchange asset ID for the asset to deposit.` since no deposit is occurring in these two functions -- we're just referring to (a) previous deposit(s)

Same would apply for `SP2Owner.sol`